src/main/bash/sdkman-init.sh

```diff
@@ -225,8 +225,9 @@ if [[ "$sdkman_auto_env" == "true" ]]; then
 
 			export SDKMAN_OLD_PWD="$PWD"
 		}
-
-		[[ -z "$PROMPT_COMMAND" ]] && PROMPT_COMMAND="sdkman_auto_env" || PROMPT_COMMAND="${PROMPT_COMMAND%\;};sdkman_auto_env"
+		
+		TRIMMED_PROMPT_COMMAND="${PROMPT_COMMAND%%*( )}"
```

Comment:
Does this parameter expansion really work? I've tested it on a few example strings and it didn't remove trailing spaces.

Reply:
I've tested it using bash. Tomorrow I will check if extglob is on, but I have default config of Ubuntu.

I will also prepare some tests to easy run for example in docker.

Maybe I should use another method like sed/awk, but I wanted to have minimal dependencies. If you prefere this solution i would Change my commit.

Moment ago i've read about xargs but it removes  double spaces also in the middle of text so it does too many for our needs.

Reply:
This should work:

```
trimmed_prompt_command="${PROMPT_COMMAND%"${PROMPT_COMMAND##*[![:space:]]}"}"
```

(taken straight from the Bash Bible: https://github.com/dylanaraps/pure-bash-bible)

Reply:
I used your fragment.
Default in Ubuntu extglob is on.

Prepare some test script, and it works:
```bash
#!/bin/bash
test_prompt () {
	test_prompt=$1
	trimmed_prompt_command="${test_prompt%"${test_prompt##*[![:space:]]}"}"
	[[ -z "$trimmed_prompt_command" ]] && test_prompt="sdkman_auto_env" || test_prompt="${trimmed_prompt_command%\;};sdkman_auto_env"
	echo $test_prompt
}

# expected output for each input: aaa;sdkman_auto_env
test_prompt "aaa"
test_prompt "aaa "
test_prompt "aaa;"
test_prompt "aaa; "
```

Of course i've tested change in commit with my local sdkman and installed hstr application (root cause of this issue).